### ISSUE TITLE
Check should be able to recognize 429 error and fail fast

### ISSUE THREAD

**Issue by xiaohansong** [airbytehq/airbyte#17980]
https://github.com/airbytehq/oncall/issues/759

Currently I observed that during the check phase of a sync job if a source reported 429 error (too many requests) it either sleeps or retries (but never succeeds). Eventually the temporal activity would time out and thus it fails, but reporting temporal issue instead.

I think since this is a data source issue we should let customer know clearly that they are facing 429 error and should adjust limit on their end to fix the problem.

**Comment by malikdiarra** [comment]
@xiaohansong do you have a list of the connection you checked somewhere? I want to see if some connectors are more affected than others.

**Comment by xiaohansong** [comment]
Yes - referred from metabase: https://airbyte.metabaseapp.com/question/381-detail-list-of-platform-failure?detail_date_range=past7days&connector_deployment_type=cloud&connector_deployment_type=internal-cloud&connector_deployment_type=cloud-inactive
